sql/analyzer/replace_order_by_distance.go

```diff
@@ -0,0 +1,136 @@
+package analyzer
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+)
+
+// replaceIdxSort applies an IndexAccess when there is an `OrderBy` over a prefix of any columns with Indexes
+func replaceIdxOrderByDistance(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.Scope, sel RuleSelector, qFlags *sql.QueryFlags) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
looks fine as a first pass, but i imagine this might impact perf. A lot of expensive subcalls made for queries with ORDER BY. My first thought would be to maybe layer this into `costedIndexScans`, potentially link the LIMIT clause count in `sql.QueryFlags`, somehow consolidate Sort dropping with the other places we do that

Reply:
This pass was modeled after `replaceIdxSort`. If this is expensive for ORDER BY queries, then so is that one.

I like the idea of layering both of these into `costedIndexScans`, but that should probably be a separate PR.

I'll keep an eye on perf.